cedar-policy-core/src/evaluator.rs

```diff
@@ -91,18 +92,47 @@ impl<'e> RestrictedEvaluator<'e> {
     /// Interpret a `RestrictedExpr` into a `Value` in this evaluation environment.
     ///
     /// May return an error, for instance if an extension function returns an error
-    pub fn partial_interpret(&self, e: BorrowedRestrictedExpr<'_>) -> Result<PartialValue> {
+    pub fn partial_interpret(&self, expr: BorrowedRestrictedExpr<'_>) -> Result<PartialValue> {
         stack_size_check()?;
 
-        match e.as_ref().expr_kind() {
+        let res = self.partial_interpret_internal(&expr);
+
+        // set the returned value's source location to the same source location
+        // as the input expression had.
+        // we do this here so that we don't have to set/propagate the source
+        // location in every arm of the big `match` in `partial_interpret_internal()`.
+        // also, if there is an error, set its source location to the source
+        // location of the input expression as well, unless it already had a
+        // more specific location
+        res.map(|pval| pval.with_maybe_source_loc(expr.source_loc().cloned()))
+            .map_err(|err| match err.source_loc() {
+                None => err.with_maybe_source_loc(expr.source_loc().cloned()),
```

Comment:
It's possible no errors do.  Even if that's the case, I'm inclined to keep the code this way, for symmetry with the code in `Evaluator::partial_interpret()`, and in case new errors get added to `RestrictedEvaluator::partial_interpret_internal()` in the future?  What do you think?